types/datum.go

```diff
@@ -1500,6 +1500,9 @@ func (d *Datum) convertToMysqlEnum(sc *stmtctx.StatementContext, target *FieldTy
 		e, err = ParseEnum(target.Elems, d.GetString(), target.Collate)
 	case KindMysqlEnum:
 		e, err = ParseEnum(target.Elems, d.GetMysqlEnum().Name, target.Collate)
+		if err == nil && d.i == 0 {
```

Comment:
actually `value=0 name=""` is a truncated enum (Not valid one), maybe we can just it before parserEnum, and skip parse process